src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -320,13 +362,12 @@ protected override void ApplyParseOptionsChanged(ProjectId projectId, ParseOptio
             string newVersion = parseOptionsService.GetLanguageVersion(options);
 
             GetProjectData(projectId, out var hostProject, out var hierarchy, out var project);
-            foreach (string configurationName in (object[])project.ConfigurationManager.ConfigurationRowNames)
+            foreach (Configuration configuration in project.ConfigurationManager)
             {
                 switch (hostProject.Language)
                 {
                     case LanguageNames.CSharp:
-                        var csharpProperties = (VSLangProj80.CSharpProjectConfigurationProperties3)project.ConfigurationManager
-                            .ConfigurationRow(configurationName).Item(1).Object;
+                        var csharpProperties = (VSLangProj80.CSharpProjectConfigurationProperties3)configuration.Object;
```

Comment:
Awesome. Thanks for fixing that!
It looks like we don't have test automation verifying the updated csproj file. Can you verify this manually on a file with (initially) different LangVersions in different configuration?

Reply:
@jcouv Yes, it works correctly if I try manually. But I don't know how this could be tested :/ GetProjectData doesn't work in a test environment

Reply:
Actually, I don't even know if it's possible to get a VisualStudioWorkspace in a test environment.

Reply:
We would need to write an integration test for this.   That said, this is a case where we really should do just that.  This is something that could easily break in the future and no-one might notice.  An integration test here would likely be a PITA, but woudl also be very important.

@jcouv or @jasonmalinowski can you help @Neme12  out with this?

Reply:
I've been trying to write an integration test but can't make a project with a custom csproj (so that it can have more than 1 platform).

Reply:
I have to defer to @jasonmalinowski for how to best test this.

Reply:
Thanks, I found a way to do this already but had to make a new helper method.

Reply:
Tests added.

Reply:
Yes, we'll want an integration test here.

> I've been trying to write an integration test but can't make a project with a custom csproj (so that it can have more than 1 platform).

I wouldn't worry about that -- let's make sure the simple case works at all before trying to get fancy.

Reply:
@jasonmalinowski If we don't have a custom .csproj and use one from a template, it will be far from simple but more importantly, it would break any time the default template is changed. Note: I just added those tests with a custom .csproj

Reply:
I don't think we would ever change teh default .csproj to include "allow unsafe" being on by default.  So i don't think that's something we really need to care about here.  And, if we are really concerned about htat, the test can check initially that the project does not have "allow unsafe" 'on' and throw if it does.  That way, we'll catch it if that ever were to change.

Reply:
@Neme12 I guess I'd prefer we didn't have an exact assertion for the file contents, but just that it contains the tags we expected to be added. Seems less brittle, otherwise any other change to the project system that changes how they stick whitespace or format something can break us.

Reply:
@CyrusNajmabadi That's not what I mean. The test I added creates a minimal csproj and compares the updated csproj with an expected one. That would break if I used a template one as an input and the template changed.

Reply:
> I guess I'd prefer we didn't have an exact assertion for the file contents, but just that it contains the tags we expected to be added.

➡️ Fixed in 7003d31fd2d7280f3f70cd41cd84f2258ffe288a